functions/nfts/asset/[[index]].ts

```diff
@@ -9,11 +9,20 @@ export const onRequest: PagesFunction = async ({ params, request, next }) => {
   const assetPromise = getAsset(collectionAddress, tokenId, request.url)
   const resPromise = next()
   try {
-    const [data, res] = await Promise.all([assetPromise, resPromise])
-    if (!data) {
-      return resPromise
+    const cache = caches.default
+    const response = await cache.match(request.url)
```

Comment:
Will the cache clear when we release a new version? Will we need to always get a response from cloudflare? should we only be caching the metadata?

Reply:
I don't believe the cache is cleared upon a new release.

I assume we do always want to get a response from cloudflare as it ideally should be faster to just get the response from the cloudflare cache and return it rather than building our own. The only situation this would not be ideal for is if its a first load, but I can't think of a way to determine that without eliciting a response from cloudflare.

Because the cache doesn't clear upon new version release, I do think it is best if we only cache the metadata as if we cache the entire response, we might serve an older version.